packages/editor/src/components/post-locked-modal/index.js

```diff
@@ -154,11 +165,10 @@ function PostLockedModal() {
 		return null;
 	}
 
-	// Avoid sending the modal if sync is supported, but retain functionality around locks etc.
-	if ( window.__experimentalEnableSync && supportsSync ) {
-		if ( globalThis.IS_GUTENBERG_PLUGIN ) {
-			return null;
-		}
+	// If both the current user and the lock owner can collaboratively edit, do not
+	// show the modal.
+	if ( syncEnabled && user.syncEnabled ) {
```

Comment:
I wonder if we could have Gutenberg augment this property on the user or even here? We could have a filter that allows it to be overriden by a plugin?

I do get that this needs to be a change in core if we want to merge this back into trunk, hence why I'm trying to see if there is a path forward.

Reply:
Yes, we could filter in 2-3 spots for the same effect. Still unlikely to be merged IMO.